x-pack/plugins/ml/public/application/explorer/swimlane_annotation_container.tsx

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC, useEffect } from 'react';
+import d3 from 'd3';
+import { scaleTime } from 'd3-scale';
+import { i18n } from '@kbn/i18n';
+import { formatHumanReadableDateTimeSeconds } from '../../../common/util/date_utils';
+import { AnnotationsTable } from '../../../common/types/annotations';
+import { ChartTooltipService } from '../components/chart_tooltip';
+
+const ANNOTATION_HEIGHT = 12;
+export const Y_AXIS_LABEL_WIDTH = 170;
+export const Y_AXIS_LABEL_PADDING = 8;
+export const Y_AXIS_LABEL_FONT_COLOR = '#6a717d';
+
+interface SwimlaneAnnotationContainerProps {
+  chartWidth: number;
+  domain: {
+    min: number;
+    max: number;
+  };
+  annotationsData?: AnnotationsTable['annotationsData'];
+  tooltipService: ChartTooltipService;
+}
+
+export const SwimlaneAnnotationContainer: FC<SwimlaneAnnotationContainerProps> = ({
+  chartWidth,
+  domain,
+  annotationsData,
+  tooltipService,
+}) => {
+  const canvasRef = React.useRef<HTMLDivElement | null>(null);
+
+  useEffect(() => {
+    if (canvasRef.current !== null && Array.isArray(annotationsData)) {
+      const chartElement = d3.select(canvasRef.current);
+      chartElement.selectAll('*').remove();
+
+      const startingXPos = Y_AXIS_LABEL_WIDTH + 2 * Y_AXIS_LABEL_PADDING;
+      const endingXPos = startingXPos + chartWidth - Y_AXIS_LABEL_PADDING;
```

Comment:
Does the end of the annotation chart need adjusting? It always seems to extend past the swim lane for me:

![image](https://user-images.githubusercontent.com/7405507/115220744-d1fca700-a100-11eb-8157-1da8d571e088.png)